### PR TITLE
fix(javascript): allow async param on createIterablePromise

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Allow all params of `createIterablePromise` to return `PromiseLike` and call them with await, if people want to use `await` inside their fonctions.

This is not a breaking change and stays compatible with before.

### CODE REVIEW

**Review by shortcuts — COMMENTED** [review]
great find! can we leave one of the playground with an async and sync implem in it? just so the build secures it

**Comment by shortcuts** [comment]
> great find! can we leave one of the playground with an async and sync implem in it? just so the build secures it

actually just a test in client-common would be good?

File: clients/algoliasearch-client-javascript/packages/client-common/tsconfig.json [hunk 1]
```diff
@@ -2,8 +2,9 @@
   "extends": "../../tsconfig.json",
   "compilerOptions": {
     "types": ["node", "vitest/globals"],
-    "outDir": "dist"
+    "outDir": "dist",
+    "skipLibCheck": true
```
Comment (shortcuts) [inline clients/algoliasearch-client-javascript/packages/client-common/tsconfig.json:6]:
```suggestion
    "outDir": "dist"
```

does it still work without it?

**Review by shortcuts — COMMENTED** [review]
:noice:

File: clients/algoliasearch-client-javascript/packages/client-common/tsconfig.json (hunk 1 above)
Comment (millotp) [reply clients/algoliasearch-client-javascript/packages/client-common/tsconfig.json:6]: nope, still complains about rollup

Comment (shortcuts) [reply clients/algoliasearch-client-javascript/packages/client-common/tsconfig.json:6]: (╯°□°)╯︵ dnʅʅoɹ

**Review by shortcuts — APPROVED** [review]
looks bon to moi gg!

**Pull request merged**